rust/arrow/src/compute/kernels/cast.rs

```diff
@@ -1480,6 +1498,19 @@ mod tests {
         assert!(c.is_null(2));
     }
 
+    #[test]
+    fn test_cast_string_to_timestamp() {
+        let a = StringArray::from(vec![Some("2020-09-08T12:00:00+00:00"), None]);
```

Comment:
I recommend adding an error case here too -- like `Some("not a date")` to ensure / encode the make-a-null-if-invalid-parse behavior

Reply:
Sure, good catch 👍